src/registration/Register.tsx

```diff
@@ -0,0 +1,199 @@
+      setBusy(true);
+      const orgFields = partOfOrganization
+        ? {
+            institute,
+            department,
+            title,
+          }
+        : {};
+      await Ajax().User.profile.set({
+        firstName: givenName,
+        lastName: familyName,
+        contactEmail: email,
+        interestInTerra,
+        ...orgFields,
+      });
+      await Ajax().User.setUserAttributes({ marketingConsent });
+      authStore.update((state) => ({ ...state, registrationStatus: 'registeredWithoutTos' }));
+      await refreshTerraProfile();
+      await refreshSamUserAttributes();
+      Ajax().Metrics.captureEvent(Events.user.register);
+    } catch (error) {
+      reportError('Error registering', error);
+      setBusy(false);
+    }
+  };
+  const errors = validate(
+    { givenName, familyName, email, institute, title, department },
+    constraints({ partOfOrganization })
+  );
+
+  return (
+    <div
+      role="main"
+      style={{
+        flexGrow: 1,
+        padding: '5rem',
+        backgroundImage: `url(${planet})`,
+        backgroundRepeat: 'no-repeat',
+        backgroundSize: '750px',
+        backgroundPosition: 'right 0px bottom -600px',
+      }}
+    >
+      <RegistrationLogo />
+      <h1
+        style={{
+          marginTop: '4rem',
+          color: colors.dark(0.6),
+          fontSize: '1.5rem',
+          fontWeight: 500,
+        }}
+      >
+        New User Registration
+      </h1>
+      <div style={{ marginTop: '1rem', display: 'flex', lineHeight: '170%' }}>
+        <LabelledTextInput
+          required
+          value={givenName}
+          onChange={setGivenName}
+          inputStyle={{ display: 'block' }}
+          label="First Name"
+        />
+        <LabelledTextInput
+          value={familyName}
+          onChange={setFamilyName}
+          inputStyle={{ display: 'block' }}
+          label="Last Name"
+        />
+      </div>
+      <div style={{ lineHeight: '170%' }}>
+        <LabelledTextInput
+          value={email}
+          required
+          onChange={setEmail}
+          labelStyle={{ display: 'block', marginTop: '2rem' }}
+          inputStyle={{ width: '66ex' }}
+          label="Contact Email for Notifications"
+        />
+        <LabelledTextInput
+          value={institute}
+          required={partOfOrganization}
+          disabled={!partOfOrganization}
+          onChange={setInstitute}
+          inputStyle={{ width: '66ex' }}
+          label="Organization"
+        />
+      </div>
+      <div style={{ lineHeight: '170%', marginTop: '0.25rem' }}>
+        <LabeledCheckbox
+          checked={!partOfOrganization}
+          onChange={() => setPartOfOrganization(!partOfOrganization)}
+          disabled={false}
+        >
+          <span style={{ marginLeft: '0.25rem' }}>I am not a part of an organization</span>
+        </LabeledCheckbox>
+      </div>
+      <div style={{ display: 'flex', lineHeight: '170%' }}>
+        <LabelledTextInput
```

Comment:
There's some duplication here that I think could be reduced by making this:
```
_.map(organizationInput => <LabelledTextInput value={value} required={partOfOrganization}... /> , [{value: department, onChange: setDepartment, label: "Department"}, {value: title, onChange: setTitle, label: "Title"}])
```

There might be good other opportunities for that sort of thing above, I was seeing a lot of duplication within blocks.